src/lib/RedisCache.ts

```diff
@@ -169,6 +177,16 @@ export class RedisCache extends CacheInstance {
       return false;
     }
 
+    if (value.startsWith(RedisCache.ERROR_PREFIX)) {
+      const deserializedError = JSON.parse(value.substring(RedisCache.ERROR_PREFIX.length));
+      const error = new Error(deserializedError.message);
+      // restore potential Error metadata set as object properties
+      for (const [k, v] of Object.entries(deserializedError)) {
+        error[k] = v;
+      }
+      return error;
```

Comment:
I believe you can simply do:
```suggestion
      return Object.assign(new Error(), deserializedError);
```